R/report_null.R

```diff
@@ -0,0 +1,71 @@
+#' Creates `NULL` Report
+#'
+#' @param tlg to convert to null report.
+#' @param ind (`integer`) indentation for the outputs of class `VTableTree`.
+#' @param ... not used.
+#'
+#' @rdname report_null
+#' @name report_null
+#'
+#' @returns the `tlg` object or a `NULL` report if the `tlg` is `NULL`, is a `TableTree` with 0 rows, is a `listing_df`
+#'   with 0 rows or is a `list` with 0 elements.
+#'
+#' @export
+#' @examples
+#' report_null(NULL)
+setGeneric("report_null", function(tlg, ...) {
+  standardGeneric("report_null")
+})
+
+#' @rdname report_null
+setMethod("report_null", "NULL", function(tlg, ind = 2L, ...) {
+  res <- null_report
+  table_inset(res) <- ind
+  res
+})
+
+#' @rdname report_null
+setMethod("report_null", "VTableTree", function(tlg, ind = 2L, ...) {
+  res <- if (nrow(tlg) == 0L || count_children(tlg) == 0) {
+    null_report
+  } else {
+    tlg
+  }
+
+  table_inset(res) <- ind
```

Comment:
is this always overloaded then?

Reply:
We import the entire `rtables` package in `chevron`.

Reply:
I meant that it is a global default that you cannot change? so all tables have inset right?